nixos/modules/services/matrix/matrix-authentication-service.nix

```diff
@@ -0,0 +1,443 @@
+      '';
+      type = types.submodule {
+        freeformType = format.type;
+
+        options = {
+          http.public_base = mkOption {
+            type = types.str;
+            default = "http://[::]:8080/";
+            description = ''
+              Public URL base used when building absolute public URLs.
+            '';
+          };
+          http.issuer = mkOption {
+            type = types.str;
+            default = "http://[::]:8080/";
+            description = ''
+              OIDC issuer advertised by the service.
+            '';
+          };
+          http.trusted_proxies = mkOption {
+            type = types.listOf (types.str);
+            default = [
+              "127.0.0.1/8"
+              "::1/128"
+            ];
+            description = ''
+              MAS can infer the client IP address from the X-Forwarded-For header. It will trust the value for this header only if the request comes from a trusted reverse proxy listed here.
+            '';
+          };
+          http.listeners = mkOption {
+            type = types.listOf (
+              types.submodule {
+                freeformType = format.type;
+                options = {
+                  name = mkOption {
+                    type = types.str;
+                    example = "web";
+                    description = ''
+                      The name of the listener, used in logs and metrics.
+                    '';
+                  };
+                  proxy_protocol = mkOption {
+                    type = types.bool;
+                    description = ''
+                      Whether to enable the PROXY protocol on the listener.
+                    '';
+                  };
+                  resources = mkOption {
+                    type = types.listOf (
+                      types.submodule {
+                        freeformType = format.type;
+                        options = {
+                          name = mkOption {
+                            type = types.str;
+                            description = ''
+                              Serve the given folder.
+                            '';
+                          };
+                          path = mkOption {
+                            type = types.str;
+                            default = "";
+                            description = ''
+                              Serve the folder on the given path.
+                            '';
+                          };
+                        };
+                      }
+                    );
+                    description = ''
+                      List of resources to serve.
+                    '';
+                  };
+                  binds = mkOption {
+                    type = types.listOf (
+                      types.submodule {
+                        freeformType = format.type;
+                        options = {
+                          host = mkOption {
```

Comment:
Would be great to have the socket option here (https://element-hq.github.io/matrix-authentication-service/reference/configuration.html#httplisteners)